src/crimson/osd/ops_executer.cc

```diff
@@ -970,20 +970,28 @@ std::unique_ptr<OpsExecuter::CloningContext> OpsExecuter::execute_clone(
   osd_op_params->at_version.version++;
   encode(cloned_snaps, cloning_ctx->log_entry.snaps);
 
-  // update most recent clone_overlap and usage stats
-  assert(cloning_ctx->new_snapset.clones.size() > 0);
-  // In classic, we check for evicted clones before
-  // adjusting the clone_overlap.
-  // This check is redundant here since `clone_obc`
-  // was just created (See prepare_clone()).
-  interval_set<uint64_t> &newest_overlap =
-    cloning_ctx->new_snapset.clone_overlap.rbegin()->second;
-  osd_op_params->modified_ranges.intersection_of(newest_overlap);
-  delta_stats.num_bytes += osd_op_params->modified_ranges.size();
-  newest_overlap.subtract(osd_op_params->modified_ranges);
   return cloning_ctx;
 }
 
+void OpsExecuter::update_clone_overlap() {
+  if (op_info.may_write() && obc->obs.exists && !snapc.snaps.empty()) {
```

Comment:
This condition is equal to `op_info.may_write() && should_clone(*obc, snapc)` which is already checked in OpsExecuter's ctor. If the condition is true, `cloning_ctx` should have a value (non-null).
`OpsExecuter::flush_clone_metadata` checks `if (cloning_ctx)` - IIUC we should call `update_clone_overlap` there.

Reply:
this condition not equal to `op_info.may_write() && should_clone(*obc, snapc)`, this condition not contains `snapc.snaps[0] > initial_obc.ssc->snapset.seq`